src/Core/Managed/Net45/Extensibility/Implementation/RichPayloadEventSource.cs

```diff
@@ -0,0 +1,498 @@
+                        var data = telemetryItem.Data;
+                        extendedData = new
+                        {
+                            PartA_iKey = telemetryItem.Context.InstrumentationKey,
+                            PartA_Tags = telemetryItem.Context.Tags,
+                            PartB_RemoteDependencyData = new { data.async, data.commandName, data.data, data.dependencyKind, data.dependencySource, data.dependencyTypeName, data.duration, data.id, data.kind, data.measurements, data.name, data.properties, data.resultCode, data.success, data.target, data.value, data.ver }
+                        };
+                        break;
+                    }
+
+                case MetricTelemetry.TelemetryName:
+                    {
+                        var telemetryItem = item as MetricTelemetry;
+                        var data = telemetryItem.Data;
+                        extendedData = new
+                        {
+                            PartA_iKey = telemetryItem.Context.InstrumentationKey,
+                            PartA_Tags = telemetryItem.Context.Tags,
+                            PartB_MetricData = new
+                            {
+                                metrics = data.metrics.Select(i => new { i.count, i.kind, i.max, i.min, i.name, i.stdDev, i.value }),
+                                data.properties,
+                                data.ver
+                            }
+                        };
+                        break;
+                    }
+
+                case ExceptionTelemetry.TelemetryName:
+                    {
+                        var telemetryItem = item as ExceptionTelemetry;
+                        var data = telemetryItem.Data;
+                        extendedData = new
+                        {
+                            PartA_iKey = telemetryItem.Context.InstrumentationKey,
+                            PartA_Tags = telemetryItem.Context.Tags,
+                            PartB_ExceptionData = new
+                            {
+                                exceptions = data.exceptions.Select(i => new
+                                {
+                                    i.hasFullStack,
+                                    i.id,
+                                    i.message,
+                                    i.outerId,
+                                    parsedStack = i.parsedStack.Select(j => new { j.assembly, j.fileName, j.level, j.line, j.method }),
+                                    i.stack,
+                                    i.typeName
+                                }),
+                                data.handledAt,
+                                data.measurements,
+                                data.problemId,
+                                data.properties,
+                                data.severityLevel,
+                                data.ver
+                            }
```

Comment:
If you can move generation of every anonymous class into separate file - we can generate those from the bond schema in future. The similar way we generate data classes today: https://github.com/Microsoft/ApplicationInsights-dotnet/blob/publicSchemaUpdate/Schema/generateSchema.ps1